google-cloud-testing/spanner-benchwrapper/src/main/java/com/google/cloud/benchwrapper/SpannerBenchWrapperImpl.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.benchwrapper;
+
+import static com.google.cloud.spanner.TransactionRunner.TransactionCallable;
+
+import com.google.api.core.ApiFunction;
+import com.google.cloud.benchwrapper.SpannerBenchWrapperGrpc.SpannerBenchWrapperImplBase;
+import com.google.cloud.spanner.DatabaseClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.Mutation;
+import com.google.cloud.spanner.ReadOnlyTransaction;
+import com.google.cloud.spanner.ResultSet;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerBatchUpdateException;
+import com.google.cloud.spanner.SpannerOptions;
+import com.google.cloud.spanner.Statement;
+import com.google.cloud.spanner.TransactionContext;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.stub.StreamObserver;
+import java.util.ArrayList;
+import java.util.List;
+
+class SpannerBenchWrapperImpl extends SpannerBenchWrapperImplBase {
+  private Spanner spanner;
+  private DatabaseClient dbClient;
+
+  public SpannerBenchWrapperImpl() {
+    SpannerOptions.Builder builder =
+        SpannerOptions.newBuilder()
+            .setProjectId("test-project-id")
+            // Channels are secure by default (via SSL/TLS). For the example we disable TLS to avoid
+            // needing certificates.
```

Comment:
We might need to move this logic to the client library as well. Otherwise by setting SPANNER_EMULATOR_HOST, it wouldn't work directly with the client library if TLS is not disabled.

Reply:
Thanks for pointing this out. Fixed.